include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -0,0 +1,840 @@
+                        {
+                            __sub_group_partials[__carry_offset] =
+                                __reduce_op(__carry_last.__v, __sub_group_partials[__carry_offset]);
+                            __carry_offset += __sub_group_size;
+                        }
+                        if (__sub_group_local_id == 0)
+                            __sub_group_partials[__active_subgroups] = __carry_last.__v;
+                        __carry_last.__destroy();
+                    }
+                }
+
+                __dpl_sycl::__group_barrier(__ndi);
+
+                // Get inter-work group and adjusted for intra-work group prefix
+                bool __sub_group_carry_initialized = true;
+                if (__block_num == 0)
+                {
+                    if (__sub_group_id > 0)
+                    {
+                        _InitValueType __value = __sub_group_id - 1 < __active_subgroups
+                                                     ? __sub_group_partials[__sub_group_id - 1]
+                                                     : __sub_group_partials[__active_subgroups - 1];
+                        oneapi::dpl::unseq_backend::__init_processing<_InitValueType>{}(__init, __value, __reduce_op);
+                        __sub_group_carry.__setup(__value);
+                    }
+                    else if (__group_id > 0)
+                    {
+                        _InitValueType __value = __sub_group_partials[__active_subgroups];
+                        oneapi::dpl::unseq_backend::__init_processing<_InitValueType>{}(__init, __value, __reduce_op);
+                        __sub_group_carry.__setup(__value);
+                    }
+                    else // zeroth block, group and subgroup
+                    {
+                        if constexpr (std::is_same_v<_InitType,
+                                                     oneapi::dpl::unseq_backend::__no_init_value<_InitValueType>>)
+                        {
+                            // This is the only case where we still don't have a carry in.  No init value, 0th block,
+                            // group, and subgroup. This changes the final scan through elements below.
+                            __sub_group_carry_initialized = false;
+                        }
+                        else
+                        {
+                            __sub_group_carry.__setup(__init.__value);
+                        }
+                    }
+                }
+                else
+                {
+                    if (__sub_group_id > 0)
+                    {
+                        _InitValueType __value = __sub_group_id - 1 < __active_subgroups
+                                                     ? __sub_group_partials[__sub_group_id - 1]
+                                                     : __sub_group_partials[__active_subgroups - 1];
```

Comment:
Same comment: min operation
```suggestion
                        _InitValueType __value = __sub_group_partials[std::min( __sub_group_id - 1, __active_subgroups - 1)];
```